lib/generators/shopify_app/install/templates/shopify_app.rb.tt

```diff
@@ -1,17 +1,18 @@
-unless defined? Rails::Generators
-  ShopifyApp.configure do |config|
-    config.application_name = "<%= @application_name %>"
+ShopifyApp.configure do |config|
+  config.application_name = "<%= @application_name %>"
+  config.old_secret = "<%= @old_secret %>"
+  config.scope = "<%= @scope %>" # Consult this page for more scope options:
+                                  # https://help.shopify.com/en/api/getting-started/authentication/oauth/scopes
+  config.embedded_app = <%= embedded_app? %>
+  config.after_authenticate_job = false
+  config.api_version = "<%= @api_version %>"
+  config.shop_session_repository = 'Shop'
+  config.allow_jwt_authentication = <%= !with_cookie_authentication? %>
+  config.allow_cookie_authentication = <%= with_cookie_authentication? %>
+
+  if defined?(Rails::Server) || defined?(Rails::Console) || (File.basename($PROGRAM_NAME) === 'rake')
```

Comment:
Since Rails 5, some `rake` commands are also callable with `rails`, e.g. `rails routes` instead of `rake routes` – can we check this is handled correctly?

Reply:
Also, the comparison can be `==`, the `===` is very rarely needed in Ruby.

Reply:
I'm starting to wonder if the `raise` call is worth going through all of this trouble. We could simply let the keys be empty if the value isn't set rather than checking for all possible scenarios.

What if we just log something if `Server` is defined and let things slide with an empty key for other cases?

Reply:
Starting the server with missing env vars would be the most common mistake. so how about we raise on that, but ignore the others?

Reply:
Excellent point, I agree. Making it happen 🙂